sdk/appconfiguration/app-configuration/test/public/featureFlag.spec.ts

```diff
@@ -185,4 +185,37 @@ describe("AppConfigurationClient - FeatureFlag", () => {
       await client.deleteConfigurationSetting({ key: secondSetting.key });
     });
   });
+
+  describe("serializeAsConfigurationSettingParam", () => {
+    let client: AppConfigurationClient;
+    let recorder: Recorder;
+
+    beforeEach(async function(this: Context) {
+      recorder = startRecorder(this);
+      client = createAppConfigurationClientForTests() || this.skip();
+    });
+
+    afterEach(async function(this: Context) {
+      await recorder.stop();
+    });
+
+    [`[]`, "Hello World"].forEach((value) => {
+      it(`Unexpected value ${value} as feature flag value`, async () => {
+        const featureFlag: ConfigurationSetting<FeatureFlagValue> = {
+          contentType: featureFlagContentType,
+          key: `${featureFlagPrefix}${recorder.getUniqueName("name-1")}`,
+          isReadOnly: false,
+          value: { conditions: { clientFilters: [] }, enabled: true }
+        };
+        featureFlag.value = value as any;
+        await client.addConfigurationSetting(featureFlag);
+        assert.equal(
+          (await client.getConfigurationSetting({ key: featureFlag.key })).value,
+          value as any,
+          "message"
+        );
+        await client.deleteConfigurationSetting({ key: featureFlag.key });
```

Comment:
Just to note - I have done a bad job in the code base of properly cleaning up settings.

With that said, you'd want to do this in an afterEach() or similar to do it correctly. If your test fails it'll skip over your cleanup altogether.

Reply:
> Just to note - I have done a bad job in the code base of properly cleaning up settings.

Nop, all the regular tests are fine. This was a separate test suite for the failure cases.
Doesn't matter. Updated to what you have suggested anyways.